microcosm/tests/config/test_validation.py

```diff
@@ -0,0 +1,79 @@
+"""
+Test validation.
+
+"""
+from hamcrest import (
+    assert_that,
+    calling,
+    has_entries,
+    raises,
+)
+from microcosm.api import binding, defaults
+from microcosm.config.api import boolean, configure, required, ValidationError
+from microcosm.loaders import load_from_dict
+from microcosm.metadata import Metadata
+from microcosm.registry import Registry
+
+
+class TestValidation:
+
+    def setup(self):
+        self.metadata = Metadata("test")
+        self.registry = Registry()
+
+        @binding("foo", registry=self.registry)
+        @defaults(
+            value=required(int),
```

Comment:
For some reason I still find using `value` as a key really confusing. 

I think what this code says is that there is a key called "value" in the `foo` namespace which is required and should be treated as an int. Calling it `value` makes me feel like you're trying to say something about the `value` of `foo` in the graph itself. I don't think you're saying that. Is that right?
 
```
@defaults(
  bar=required(int),
)
```

Reply:
Would you feel the same if we called it `username` or `password`? It's still a _label_ on the LHS for the value on the RHS.